File: math/broken-calculator/broken-calculator.js
/**
 * @param {number} X
 * @param {number} Y
 * @return {number}
 */
function brokenCalc(X, Y) {
  let ops = 0;

  while (X < Y) {
    if (Y % 2 === 1) {
      Y++;
    } else {
      Y /= 2;
    }
    ops++;
  }

  return X - Y + ops;
}

File: math/broken-calculator/broken-calculator.go
package brokencalculator

func brokenCalc(X int, Y int) int {
	ops := 0

	for X < Y {
		if Y%2 == 1 {
			Y++
		} else {
			Y /= 2
		}

		ops++
	}

	return X - Y + ops
}
